### PR TITLE
47641154 - COMP3710 - Mask R-CNN for Lesion Detection

### PR DESCRIPTION
**Pull request by that1potato**
Hi,

Here is my Mask R-CNN solution for the 3rd task in assignment 3.

You can see more details in README.

Please review and let me know if any changes need to be made to suit you.

Request by
Pengfei Zhang,
Student ID: 47641154

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
- Commit messages could be more descriptive
- Readme is OK. Not much interpretations given for results achieved.
- No plots shown for training/validation results
- No performance metrics shown - IoU, mAP or classification accuracy 
- Total loss seems to fluctuate heavily, so performance seems poor
- Doesn't seem to have performed any hyperparameter tuning

**Comment by that1potato** [comment]
> - Commit messages could be more descriptive
> - Readme is OK. Not much interpretations given for results achieved.
> - No plots shown for training/validation results
> - No performance metrics shown - IoU, mAP or classification accuracy 
> - Total loss seems to fluctuate heavily, so performance seems poor
> - Doesn't seem to have performed any hyperparameter tuning

Thanks for reviewing.

What is the final deadline for me to try to fix these issues?

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem -3 (some metrics missing, loss fluctuation, poor performance)
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage -2 (no plots etc.)
Module present
Commenting
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required -2 (please remove the GCN work/readme etc., nothing further required)
Request Description good

**Comment by wotahman** [comment]
Sorry Shakes, I'm unable to remove it from the commit

**Comment by that1potato** [comment]
> ## Good Practice (Design/Commenting, TF/Torch Usage)
> Adequate use and implementation
> Good spacing and comments
> Header blocks
> ## Recognition Problem 
> Solves problem -3 (some metrics missing, loss fluctuation, poor performance)
> Driver Script present
> File structure present
> Shows Usage & Demo & Visualisation & Data usage -2 (no plots etc.)
> Module present
> Commenting
> No Data leakage
> Difficulty: Normal -5
> ## Commit Log
> Meaningful commit messages
> Progressive commits used
> ## Documentation
> ReadMe acceptable/good
> Good Description and Comments 
> Markdown used PDF submitted
> ## Pull Request
> Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
> Feedback required -2 (please remove the GCN work/readme etc., nothing further required)
> Request Description good

Sorry Shakes, I don't think I'm able to remove it from my previous commit

**Comment by shakes76** [comment]
All good, feedback attempted so marks will be awarded, but I can't merge it though.
